src/librustc/ty/maps.rs

```diff
@@ -580,10 +598,17 @@ macro_rules! define_maps {
                        key,
                        span);
 
+                profq_msg!(tcx,
+                    ProfileQueriesMsg::QueryBegin(span.clone(),
+                                                  QueryMsg::$name(format!("{:?}", key))));
```

Comment:
I'm not entirely happy with this formatting the key. I would *prefer* it if we started with just the query type, and not the key. *Also*, I would *definitely not* record individual keys *except* by having a threshold of a minimal duration that someone might be interested in. And there is a way to measure time on the spot here, that could be placed in `tcx.maps` - basically, adjust the "last query start time" after finishing an inner query so the inner time is *not* included in the outer one.